cli/src/commands/ci.rs

```diff
@@ -0,0 +1,228 @@
+use anyhow::{ensure, Context, Result};
+use clap::Parser;
+use cli_core::rivet_api::models;
+use std::collections::HashMap;
+use tokio::fs;
+
+use crate::{
+	commands::version,
+	util::{paths, term},
+};
+
+#[derive(Parser)]
+pub enum SubCommand {
+	/// Generates a workflow
+	Generate {
+		#[clap(subcommand)]
+		command: GenerateOpts,
+	},
+}
+
+impl SubCommand {
+	pub async fn execute(&self, ctx: &cli_core::Ctx) -> Result<()> {
+		match self {
+			SubCommand::Generate { command } => command.execute(ctx).await,
+		}
+	}
+}
+
+#[derive(Parser)]
+pub enum GenerateOpts {
+	#[clap(name = "github")]
+	GitHub(GenerateGitHubOpts),
+}
+
+impl GenerateOpts {
+	pub async fn execute(&self, ctx: &cli_core::Ctx) -> Result<()> {
+		match self {
+			GenerateOpts::GitHub(opts) => gen_github(ctx, opts).await,
+		}
+	}
+}
+
+#[derive(Parser)]
+pub struct GenerateGitHubOpts {
+	/// Path to generate the workflow file at
+	#[clap(long)]
+	path: Option<String>,
+
+	#[clap(long)]
+	allow_idle_lobbies: bool,
+}
+
+async fn gen_github(ctx: &cli_core::Ctx, opts: &GenerateGitHubOpts) -> Result<()> {
+	let relative_path = opts
+		.path
+		.clone()
+		.unwrap_or_else(|| ".github/workflows/rivet-deploy.yml".to_string());
+	let path = paths::project_root()?.join(relative_path);
+	let path_parent = path.parent().context("path has no parent")?;
+
+	// Create parent path for workflow
+	fs::create_dir_all(&path_parent).await?;
+
+	// Generate workflow file
+	let workflow = gen_github_workflow(ctx, opts).await?;
+
+	// Write workflow file
+	fs::write(&path, workflow).await?;
+
+	term::status::success("Config written", path.display());
+
+	Ok(())
+}
+
+// TODO: Strings in this workflow are not appropriately escaped
```

Comment:
Intentionally leaving this for now. Escaping strings for sh is not fun.

Reply:
I agree